cabal-install/src/Distribution/Client/Tar.hs

```diff
@@ -69,8 +69,12 @@ extractTarGzFile
   -- ^ Tarball
   -> IO ()
 extractTarGzFile dir expected tar =
-  Tar.unpack dir
-    . Tar.checkTarbomb expected
+  Tar.unpackAndCheck
+    ( \x ->
+        SomeException <$> Tar.checkEntryTarbomb expected x
+          <|> SomeException <$> Tar.checkEntrySecurity x
+    )
+    dir
```

Comment:
`unpackAndCheck` is a new API, unavailable from `tar-0.5` series. You can either bump lower bound to `tar >= 0.6`, or switch between `unpack` and `unpackAndCheck` with CPP, checking `#if MIN_VERSION_tar(0,6,0)`.

Reply:
I think the approach in keeping with Cabal practices would be to create Distribution.Compat.Tar and use CPP there, creating a helper function which has the same type across both versions.

Reply:
That said, could we not just call `checkSecurity` and `checkTarbomb` in succession which would let us keep making use of the old 0.5 apis that 0.6 still supports, and thus avoid the need for CPP altogether?

Reply:
@gbaz No, types are different.